tests/testProfileChangePassword.php

```diff
@@ -157,6 +159,43 @@ public function testSuccessfulPasswordChangeDuringWooAccountEdit() {
 		$this->assertEmpty( $errors->get_error_messages() );
 	}
 
+	/**
+	 * Test that the change password process handles escaped data.
+	 */
+	public function testThatPasswordIsUnescapedBeforeSending() {
+		$this->startHttpHalting();
+		self::$options->set( 'domain', 'example.auth0.com' );
+
+		$user   = $this->createUser();
+		$errors = new WP_Error();
+
+		// Store userinfo for a DB strategy user.
+		$this->storeAuth0Data( $user->ID, 'auth0' );
+
+		// Set a password with special characters.
+		$new_password        = uniqid() . '"' . uniqid() . "'" . uniqid() . '\\' . uniqid();
+		$_POST['password_1'] = wp_slash( $new_password );
+
+		// API call mocked to pass bearer check.
+		$mock_api = $this
+			->getMockBuilder( WP_Auth0_Api_Change_Password::class )
+			->setMethods( [ 'set_bearer' ] )
+			->setConstructorArgs( [ self::$options, self::$api_client_creds ] )
+			->getMock();
+		$mock_api->method( 'set_bearer' )->willReturn( true );
+		$change_password = new WP_Auth0_Profile_Change_Password( $mock_api );
+
+		$decoded_res = [];
+		try {
+			$change_password->validate_new_password( $errors, $user );
+		} catch ( Exception $e ) {
+			$decoded_res = unserialize( $e->getMessage() );
```

Comment:
the exception is present because the test is halting network activity, right?

Reply:
Correct